src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilder.cs

```diff
@@ -66,11 +69,103 @@ public async IAsyncEnumerable<ProjectBuildResult> BuildProject()
+            {
+                var projectReferencesMap = FileUtils.GetProjectsWithReferences(_workspacePath);
+
+                if (!_analyzerConfiguration.BuildSettings.SyntaxOnly)
+                {
+                    builder.Build();
+                    foreach (var projectResult in builder.Projects)
+                    {
+                        using (ProjectBuildHandler projectBuildHandler =
+                            new ProjectBuildHandler(Logger, projectResult.Project, _analyzerConfiguration))
+                        {
+                            projectBuildHandler.AnalyzerResult = projectResult.AnalyzerResult;
+                            projectBuildHandler.ProjectAnalyzer = projectResult.ProjectAnalyzer;
+                            var result = await projectBuildHandler.Build();
+                            ProjectResults.Add(result);
+                        }
+                    }
+                    if (_analyzerConfiguration.AnalyzeFailedProjects)
+                    {
+                        foreach (var projectResult in builder.FailedProjects)
+                        {
+                            using (ProjectBuildHandler projectBuildHandler =
+                            new ProjectBuildHandler(Logger, _analyzerConfiguration))
+                            {
+                                projectBuildHandler.ProjectAnalyzer = projectResult.ProjectAnalyzer;
+                                var result = projectBuildHandler.SyntaxOnlyBuild();
+                                ProjectResults.Add(result);
+                            }
+                        }
+                    }
+                } 
+                else
+                {
+                    builder.GenerateNoBuildAnalysis();
+
+                    var projectsInOrder = CreateDependencyQueue(projectReferencesMap);
+                    Dictionary<string, MetadataReference> references = new Dictionary<string, MetadataReference>();
+
+                    while (projectsInOrder.Count > 0)
+                    {
+                        var projectPath = projectsInOrder.Dequeue();
+                        var project = builder.Projects.Find(p => p.ProjectAnalyzer.ProjectFile.Path.Equals(projectPath));
+                        var projectReferencePaths = projectReferencesMap[projectPath]?.Distinct().ToHashSet<string>();
+
+                        using (ProjectBuildHandler projectBuildHandler =
+                            new ProjectBuildHandler(Logger, project.Project, _analyzerConfiguration))
+                        {
+                            projectBuildHandler.AnalyzerResult = project.AnalyzerResult;
+                            projectBuildHandler.ProjectAnalyzer = project.ProjectAnalyzer;
+                            var projectReferences = references.Where(r => projectReferencePaths.Contains(r.Key)).ToDictionary(p=>p.Key, p=> p.Value);
+                            var result = projectBuildHandler.SyntaxOnlyBuild(projectReferences);
```

Comment:
similar to above, can we add parallelism if operations are time consuming...

Reply:
Same comment. The parallelism on this is controlled from WorkspaceBuilderHelper